services/sts/src/main/java/software/amazon/awssdk/services/sts/auth/StsWebIdentityTokenFileCredentialsProvider.java

```diff
@@ -140,10 +141,16 @@ public String toString() {
     }
 
     @Override
-    protected Credentials getUpdatedCredentials(StsClient stsClient) {
+    protected SessionCredentialsHolder getUpdatedCredentials(StsClient stsClient) {
         AssumeRoleWithWebIdentityRequest request = assumeRoleWithWebIdentityRequest.get();
         notNull(request, "AssumeRoleWithWebIdentityRequest can't be null");
-        return stsClient.assumeRoleWithWebIdentity(request).credentials();
+        AssumeRoleWithWebIdentityResponse assumeRoleWithWebIdentityResponse = stsClient.assumeRoleWithWebIdentity(request);
+        return SessionCredentialsHolder.builder()
+                                       .credentials(assumeRoleWithWebIdentityResponse.credentials())
+                                       .accountId(Arn.parse(assumeRoleWithWebIdentityResponse.assumedRoleUser().arn())
```

Comment:
use accountIdFromArn from parent?

Reply:
thanks for finding this, I thought there was one more place but I couldn't find it (should've searched...)